code/modules/antagonists/_common/antag_datum.dm

```diff
@@ -65,10 +65,13 @@ GLOBAL_LIST_EMPTY(antagonists)
 	GLOB.antagonists += src
 	typecache_datum_blacklist = typecacheof(typecache_datum_blacklist)
 
+	RegisterSignal(owner, COMSIG_MIND_TRANSFERRED, .proc/on_body_removal)
```

Comment:
Shouldn’t this be on on_gain? Or apply_innate_whatever

Reply:
Yes it should